tests/components/daybetter_services/test_sensor.py

```diff
@@ -0,0 +1,219 @@
+
+    with (
+        patch(
+            "homeassistant.components.daybetter_services.DayBetterClient.fetch_sensor_data",
+            return_value=[],  # Library filters out non-sensor devices
+        ),
+        patch(
+            "homeassistant.components.daybetter_services.DayBetterClient.close",
+        ),
+    ):
+        assert await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        # No sensors should be created for non-sensor devices
+        states = hass.states.async_all()
+        sensor_states = [s for s in states if s.entity_id.startswith("sensor.")]
+        assert len(sensor_states) == 0
+
+
+async def test_sensor_update(hass: HomeAssistant) -> None:
+    """Test sensor data update."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={"token": "test_token"},
+    )
+    entry.add_to_hass(hass)
+
+    with (
+        patch(
+            "homeassistant.components.daybetter_services.DayBetterClient.fetch_sensor_data",
+            return_value=[
+                {
+                    "deviceId": "test_device_1",
+                    "deviceName": "test_sensor",
+                    "deviceGroupName": "Test Group",
+                    "deviceMoldPid": "pid1",
+                    "temp": 225,  # Raw value, will be divided by 10 to get 22.5
+                    "humi": 650,  # Raw value, will be divided by 10 to get 65.0
+                }
+            ],
+        ) as mock_fetch,
+        patch(
+            "homeassistant.components.daybetter_services.DayBetterClient.close",
+        ),
+    ):
+        assert await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        # Initial values
+        states = hass.states.async_all()
+        temp_sensors = [s for s in states if "temperature" in s.entity_id]
+        assert len(temp_sensors) == 1
+        assert temp_sensors[0].state == "22.5"
+
+        # Update with new values
+        mock_fetch.return_value = [
+            {
+                "deviceId": "test_device_1",
+                "deviceName": "test_sensor",
+                "deviceGroupName": "Test Group",
+                "deviceMoldPid": "pid1",
+                "type": 5,
+                "temp": 250,  # Raw value, will be divided by 10 to get 25.0
+                "humi": 700,  # Raw value, will be divided by 10 to get 70.0
+            }
+        ]
+
+        # Trigger coordinator refresh
+        runtime_data: DayBetterRuntimeData = hass.data[DOMAIN][entry.entry_id]
+        coordinator = runtime_data.coordinator
+        await coordinator.async_refresh()
+        await hass.async_block_till_done()
+
+        # Check updated values
+        states = hass.states.async_all()
+        temp_sensors = [s for s in states if "temperature" in s.entity_id]
+        humi_sensors = [s for s in states if "humidity" in s.entity_id]
+        assert len(temp_sensors) == 1
+        assert len(humi_sensors) == 1
```

Comment:
Use the expected entity ids instead to look up the states in the core state machine.